File: updateBudget.c
#include <stdio.h>
float income, rent, utilities, groceries, transportation, savings, spent, expenses;

void percent(char type[], int ammount){
    int per = ammount/income * 100;

    printf("your %s is %d%% of your income.\n", type, per);
}

int main(void){
    
    printf("this is a budget calculator.\n How much money do you make a month? \n");
    scanf("%f", &income);
    printf("How much do your utilities cost? \n");
    scanf("%f", &utilities);
    printf("How much does your groceries cost? \n");
    scanf("%f", &groceries);
    printf("How much do your transportation cost? \n");
    scanf("%f", &transportation);
    printf("How much do your rent cost? \n");
    scanf("%f", &rent);
    
    savings = income *.2;
    expenses = rent + utilities + groceries + transportation;
    spent = income - expenses - savings;
    
    printf("you make %.2f\n", income);
    printf("your expenses are $%.2f\n", expenses);
    printf("your savings are $%.2f\n", savings);
    printf("your spending money is $%.2f\n", spent);

    percent("rent", rent);
    percent("utilities", utilities);
    percent("groceries", groceries);
    percent("transportation", transportation);
    percent("savings", savings); 
    percent("expenses", expenses);
    percent("spending", spent);   
    return 0;
}

File: updateBudget.py
print("This is a finacial calculator")

income=input("what is your monthly income?:\n")

def user_inputs(costs):
    fees=input(f"what does your {costs} cost?:\n")
    return int(fees)
rent=user_inputs("rent")
groceries=user_inputs("groceries")
utilities=user_inputs("utilities")
transp=user_inputs("transportation")

    


savings = float(income) * float(0.2)
expenses = float(rent) + float(utilities) + float(groceries) + float(transp)
spending = float(income)-float(expenses)-float(savings)

print("your expenses are", expenses)
print("your savings is", savings)
print("your spending money is", spending)

def percent(type, ammount):
    per = ammount/income *100
    return(f"your {type} is {per}% income")


percent("rent", rent)
percent("utilities", utilities)
percent("groceries", groceries)
percent("transportation", transp)
percent("expenses", expenses)